dia-batching-server/src/api/custom/arsb.rs

```diff
@@ -0,0 +1,73 @@
+use crate::api::binance::BinancePriceApi;
+use crate::api::custom::AssetCompatibility;
+use crate::api::error::CustomError;
+use crate::types::{AssetSpecifier, Quotation};
+use async_trait::async_trait;
+use chrono::Utc;
+use rust_decimal::prelude::Zero;
+use rust_decimal::Decimal;
+
+// The blockchain and symbol for the Argentinian blue dollar.
+// These are the expected values for the asset specifier.
+pub const BLOCKCHAIN: &'static str = "FIAT";
+pub const SYMBOL: &'static str = "ARS-USD";
+
+/// Returns the price for the Argentinian blue dollar.
+/// The price is fetched from Binance as binance has a very accurate price for the blue dollar.
+pub struct ArsBluePriceView;
+
+#[async_trait]
+impl AssetCompatibility for ArsBluePriceView {
+	fn supports(&self, asset: &AssetSpecifier) -> bool {
+		asset.blockchain.to_uppercase() == BLOCKCHAIN && asset.symbol.to_uppercase() == SYMBOL
+	}
+
+	async fn get_price(&self, _asset: &AssetSpecifier) -> Result<Quotation, CustomError> {
+		ArsBluePriceView::get_price().await
+	}
+}
+
+impl ArsBluePriceView {
+	async fn get_price() -> Result<Quotation, CustomError> {
+		// Only the symbol is needed to get the price with the BinancePriceApi client.
+		let asset =
+			AssetSpecifier { blockchain: "Binance".to_string(), symbol: "USDTARS".to_string() };
+
+		let binance_price_api = BinancePriceApi::new();
```

Comment:
I noticed we are creating a new `reqwest::Client()` on each query, also I think on the others implementing `get_price`. Maybe not important for now, but if we later require higher frequency updates maybe we should clone this one and pass it around.

Reply:
I refactored it so that the clients are stored in the structs and so that there is just one instance of the CustomPriceApi.

Reply:
Great!